src/TableFunctions/CMakeLists.txt

```diff
@@ -5,5 +5,5 @@ list(REMOVE_ITEM clickhouse_table_functions_sources ITableFunction.cpp TableFunc
 list(REMOVE_ITEM clickhouse_table_functions_headers ITableFunction.h TableFunctionFactory.h)
 
 add_library(clickhouse_table_functions ${clickhouse_table_functions_sources})
-target_link_libraries(clickhouse_table_functions PRIVATE clickhouse_parsers clickhouse_storages_system dbms)
+target_link_libraries(clickhouse_table_functions PRIVATE clickhouse_parsers clickhouse_storages_system dbms Poco::MongoDB)
```

Comment:
It will also work if you move `Poco::MongoDB` from `PRIVATE` keyword section here https://github.com/ClickHouse/ClickHouse/blob/9b5cc2e7083ddfd4507b702541aaa613ebe86c9c/src/CMakeLists.txt#L395-L410  to the same place a few lines below -- under `PUBLIC` keyword.

On the other hand, with the current code, you do not need `Poco::MongoDB` linked as it is not used. So let's just remove this for now.